pkg/operator/controller/controller_lb.go

```diff
@@ -0,0 +1,161 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	ingressv1alpha1 "github.com/openshift/cluster-ingress-operator/pkg/apis/ingress/v1alpha1"
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+	"github.com/openshift/cluster-ingress-operator/pkg/util/slice"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+const (
+	// loadBalancerServiceFinalizer is applied to load balancer services to ensure
+	// we can manage deletion of associated DNS records.
+	loadBalancerServiceFinalizer = "ingress.openshift.io/operator"
+
+	// awsLBProxyProtocolAnnotation is used to enable the PROXY protocol on any
+	// AWS load balancer services created.
+	awsLBProxyProtocolAnnotation = "service.beta.kubernetes.io/aws-load-balancer-proxy-protocol"
+)
+
+// ensureLoadBalancerService creates an LB service if one is desired but absent.
+// Always returns the current LB service if one exists (whether it already
+// existed or was created during the course of the function).
+func (r *reconciler) ensureLoadBalancerService(ci *ingressv1alpha1.ClusterIngress, infra *configv1.Infrastructure, deployment *appsv1.Deployment) (*corev1.Service, error) {
+	desiredLBService, err := desiredLoadBalancerService(ci, infra, deployment)
+	if err != nil {
+		return nil, err
+	}
+
+	currentLBService, err := r.currentLoadBalancerService(ci)
+	if err != nil {
+		return nil, err
+	}
+	if desiredLBService != nil && currentLBService == nil {
+		err = r.Client.Create(context.TODO(), desiredLBService)
+		if err == nil {
+			logrus.Infof("created load balancer service %s/%s", desiredLBService.Namespace, desiredLBService.Name)
+		} else if !errors.IsAlreadyExists(err) {
+			return nil, fmt.Errorf("failed to create load balancer service %s/%s: %v", desiredLBService.Namespace, desiredLBService.Name, err)
+		}
+		return r.currentLoadBalancerService(ci)
+	}
+	return currentLBService, nil
+}
+
+// TODO: This should take operator config into account so that the operand
+// namespace isn't hard-coded.
+func loadBalancerServiceName(ci *ingressv1alpha1.ClusterIngress) types.NamespacedName {
+	return types.NamespacedName{Namespace: "openshift-ingress", Name: "router-" + ci.Name}
+}
+
+// desiredLoadBalancerService returns the desired LB service for a
+// clusteringress, or nil if an LB service isn't desired. An LB service is
+// desired if the high availability type is Cloud. An LB service will declare an
+// owner reference to the given deployment.
+func desiredLoadBalancerService(ci *ingressv1alpha1.ClusterIngress, infra *configv1.Infrastructure, deployment *appsv1.Deployment) (*corev1.Service, error) {
+	if ci.Spec.HighAvailability.Type != ingressv1alpha1.CloudClusterIngressHA {
```

Comment:
Need to check whether `ci.Spec.HighAvailability` is nil.

Reply:
Fixed